player/playback-engine/src/main/kotlin/com/tidal/sdk/player/playbackengine/player/di/MediaSourcererModule.kt

```diff
@@ -123,13 +126,34 @@ internal object MediaSourcererModule {
         .build()
 
     @Provides
-    @ExtendedExoPlayerComponent.Scoped
+    @Reusable
+    @ExtendedExoPlayerComponent.LocalWithAuth
+    fun okHttpClientWithAuth(
+        @ExoPlayerPlaybackEngineComponent.LocalWithAuth
+        okHttpClient: OkHttpClient,
+        assetTimeoutConfig: AssetTimeoutConfig,
+    ) = okHttpClient.newBuilder()
+        .connectTimeout(assetTimeoutConfig.connectTimeout.toJavaDuration())
+        .readTimeout(assetTimeoutConfig.readTimeout.toJavaDuration())
+        .writeTimeout(assetTimeoutConfig.writeTimeout.toJavaDuration())
+        .build()
+
+    @Provides
+    @ExtendedExoPlayerComponent.Local
```

Comment:
`OkHttpDataSource.Factory` was never `@Reusable`, so I didn't want to make any of these that either. `OkHttpClient` had it, so I added it to the new one there.

Maybe we **should** add it to both, but I didn't want to take any extra "risk" for this task, since changing this has nothing to do with the task at hand.

Or am I missing something?

Reply:
nah it's fine you have a point